### ISSUE TITLE
Fails to build with glibc >= 2.23

### ISSUE THREAD

**Issue by paravoid** [facebook/hhvm#6924]
This was reported as [Debian bug #818831](https://bugs.debian.org/818831): HHVM 3.12.0 fails to build against glibc 2.23 due to `isnan()`/`isinf()` incompatibilities, which seem to also be unresolved in currrent master.

[Glibc's bug #19439](https://sourceware.org/bugzilla/show_bug.cgi?id=19439) has the context for this issue. Someone at ArchLinux has [already made a fix](https://www.mail-archive.com/arch-commits@archlinux.org/msg253256.html) that I'd suggest to not read for copyright/CLA reasons :) 

The TL;DR is that in C++ programs, `#include <cmath>`  should be used instead of `#include <math.h>` and `std::isnan()` should be used instead of `isnan()` (same with `isinf()`). HHVM also uses `zend_isnan()`/`zend_isinf()` in Zend code, but this can probably go away and replaced with `std::isnan()` directly.

**Comment by Orvid** [comment]
Diff D3078029 up internally that should fix this.
